vela-templates/definitions/internal/trait/gateway.cue

```diff
@@ -9,55 +9,58 @@ gateway: {
 
 		status: {
 			customStatus: #"""
-				if context.outputs.ingress.status.loadBalancer.ingress == _|_ {
+				let nameSuffix = {
+				  if parameter.name != _|_ { "-" + parameter.name }
+				  if parameter.name == _|_ { "" }
+				}
+				let ingressMetaName = context.name + nameSuffix
+				let ig  = [for i in context.outputs if (i.kind == "Ingress") && (i.metadata.name == ingressMetaName) {i}][0]
+				let igs = ig.status.loadBalancer.ingress[0]
```

Comment:
There may be some `_|_ bottom error` if ig.status.loadBalancer is `_|_`.

Reply:
Currently, CUE can not support short-circuit in if comprehension, refer to issue: https://github.com/cue-lang/cue/issues/2232

Instead of using let, we can combine defaut value and if comprehension like:

```
igs: *null | string
if ig != _|_ if ig.status != _|_ if ig.status.loadbalancer != _|_ {
  igs: ig.status.loadbalancer.ingress[0]
}
```

Reply:
ok.  can you explain the difference in cue between `let var=` and just rendering the item as `var:` ?

Reply:
We can set default value, value or type with `:`, but only value can be set in `let`.

Reply:
So I wonder now why to ever use `let var=`?

I guess just that it doesn't appear as an output?

Reply:
Yes `let` is a declaration so the let won't be evaluated in the final result of CUE.
In your cases, we'll only lookup the value in `isHealth` and `customStatus`, so there's no differences whether you're using `let` or not.

Here's some official doc in CUE explaining let: https://cuelang.org/docs/references/spec/#let-declarations